app/services/open_library_service.rb

```diff
@@ -0,0 +1,37 @@
+class OpenLibraryService
+  cattr_accessor :base_url_endpoint
+  self.base_url_endpoint = 'https://openlibrary.org/'
+
+  def book_info(isbn)
+    @isbn = isbn.to_s.rjust(10, '0')
```

Comment:
There isn't very common to have internal state on services, it's very easy to have unexpected results if you use the same instance of the service to make multiple request. Of course you can do it if it is necessary, but I think we can address the same result here without saving the ISBN in an instance variable.

I always try to think service's methods as an orchestrator which does three things:
1) Prepares the request to be made, based on the input you give to the service
2) Makes the request
3) Parses the response and returns the result

Depending on what you have to do, and how big the project/requirement is/will be, this three things could be separated in different objects, each one with its own responsibilities. This will open the door of reusability and encapsulation, so don't limit yourself to put all the logic in the service class! Be as creative as you want and remember all you have learned in PdeP!

Said that, I will leave you another version of your code, with a little refactor to don't use any instance variable at all (I will not extract anything to another object, but keep in mind you can do it!)

```ruby
class OpenLibraryService
  BASE_URL = Rails.application.secrets.open_library[:base_url]

  def book_info(isbn)
    response = HTTParty.get(book_info_url(isbn), headers: default_headers)

    # This code isn't reusable, and we will probably want to use it somewhere else in the future
    case response.code
    when 200..299
      book_info_response(JSON.parse(response.body), isbn)
    else
      raise UnhandledResponseError, "OpenLibrey returned #{response.code} status"
    end
  end

  private

  def book_info_url(isbn)
    "#{BASE_URL}/api/books?bibkeys=ISBN:#{isbn}&format=json&jscmd=data"
  end

  def default_headers
    { 'Content-Type' => 'application/json' }
  end

  def  book_info_response(response_body, isbn)
    raise ArgumentError, 'Incorrect ISBN' unless response_body.keys.include? "ISBN:#{isbn}"
    parsed_response = response_body["ISBN:#{isbn}"].slice('title', 'subtitle', 'number_of_pages', 'authors')
    parsed_response['authors'] = parsed_response['authors']&.map { |a| a['name'] }&.compact || []

    parsed_response
  end
end
```

This solution is far from being perfect, but "does" the job and is clear enough to refactor it later if our service keeps growing. Something like this is what I expect as the minimal/simpler solution to this requirement, as you can see you weren't very far of this solution.